### ISSUE TITLE
[V2] Updating passProps to custom title component not updating the component

### ISSUE THREAD

**Issue by bragur** [wix/react-native-navigation#3782]
### Issue Description

Using `mergeOptions` to update the title component of a modal, but when `passProps` get updated the component does not receive any prop updates. If you have a glance at my code below, you'd expect the subtitle to be updated with every tick of the interval function, but it just stays the same as when initialized. Any suggestions?

### Steps to Reproduce / Code Snippets / Screenshots

```
import React, { Component } from 'react';
import { Text, View, Button } from 'react-native';
import { Navigation } from 'react-native-navigation';

export default class ModalWithTimer extends Component {
  state = {
    timer: 0,
  };

  componentWillMount() {
    this.timerId = setInterval(this.updateTimer, 1000);
    this.updateTitle();
  }

  componentWillUnmount() {
    clearInterval(this.timerId);
  }

  componentDidUpdate(_, oldState) {
    if (oldState.timer !== this.state.timer) {
      this.updateTitle();
    }
  }

  updateTimer = () => {
    this.setState({
      timer: this.state.timer + 1,
    });
  };

  updateTitle() {
    const options = {
      topBar: {
        animate: true,
        title: {
          component: {
            name: 'Title',
            passProps: {
              title: 'Modal with timer',
              subtitle: `Timer: ${this.state.timer}`,
            },
          },
        },
      },
    };

    Navigation.mergeOptions(this.props.componentId, options);
  }

  render() {
    return (
      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
        <Text> Useful Component </Text>
        <Button
          onPress={() => Navigation.dismissModal(this.props.componentId)}
          title="Dismiss me"
        />
      </View>
    );
  }
}
```

```
import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  title: {
    fontSize: 16,
    fontWeight: '600',
  },
  subtitle: {
    fontSize: 14,
  },
});

export default class TitleWithStatefulSubtitle extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{this.props.title}</Text>
        <Text style={styles.subtitle}>{this.props.subtitle}</Text>
      </View>
    );
  }
}

```
---
### Environment
* React Native Navigation version: 2.0.2478
* React Native version: 0.56.0
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator iPhone X 11.4

**Comment by luqezman** [comment]
Any updates? I have the same problem :'(

**Comment by guyca** [comment]
Will be resolved in #3030

**Comment by sophister** [comment]
same problem
